nx/include/switch/kernel/tmem.h

```diff
@@ -1,19 +1,69 @@
+/**
+ * @file tmem.h
+ * @brief Transfer memory handling
+ * @author plutoo
+ * @copyright libnx Authors
+ * @remark Transfer memory differs from shared memory in the fact that the user process (as opposed to the kernel) allocates and owns its backing memory.
+ */
 #pragma once
 #include "../types.h"
 
+/// Transfer memory information structure.
 typedef struct {
-    Handle      handle;
-    size_t      size;
-    Permission  perm;
-    void*       src_addr;
-    void*       map_addr;
+    Handle      handle;   ///< Kernel object handle.
+    size_t      size;     ///< Size of the transfer memory object.
+    Permission  perm;     ///< Permissions of the transfer memory object.
+    void*       src_addr; ///< Address of the source backing memory.
+    void*       map_addr; ///< Address to which the transfer memory object is mapped.
 } TransferMemory;
 
+/**
+ * @brief Creates a transfer memory object.
+ * @param t Transfer memory information structure that will be filled in.
+ * @param size Size of the transfer memory object to create.
+ * @param perm Permissions to assign to the transfer memory object.
+ * @return Result code.
+ */
 Result tmemCreate(TransferMemory* t, size_t size, Permission perm);
```

Comment:
Permission is the permission that it is kept as in local process (---, RW- or R--), in remote process it's always mapped RW- I think